crates/rolldown_plugin_module_federation/src/init_modules_visitor.rs

```diff
@@ -113,4 +112,16 @@ impl<'ast> VisitMut<'ast> for InitModuleVisitor<'ast, '_> {
     }
     walk_expression(self, expr);
   }
+
+  // TODO need to semantic analysis at transform_ast hook
+  fn visit_call_expression(&mut self, call_expr: &mut oxc::ast::ast::CallExpression<'ast>) {
+    if let Expression::Identifier(id) = &call_expr.callee {
+      if id.name == "require" && call_expr.arguments.len() == 1 {
```

Comment:
Here need to using semantic analysis to detect the global `require`. It need to add semantic data to `transform_ast ` hook. But it has overhead for the pipeline, @IWANABETHATGUY Other buitlin plugins used semantic data?

Reply:
Let's discuss it in next meeting.